PWGJE/Core/JetBkgSubUtils.h

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file JetBkgSubUtils.h
+/// \brief Jet background subtraction utilities
+///
+/// \author Hadi Hassan <hadi.hassan@cern.ch>, JYU
+
+#ifndef O2_ANALYSIS_JETBKGSUBUTILS_H
+#define O2_ANALYSIS_JETBKGSUBUTILS_H
+
+#include <memory>
+#include <vector>
+
+#include <TMath.h>
+
+#include "fastjet/PseudoJet.hh"
+#include "fastjet/ClusterSequenceArea.hh"
+#include "fastjet/AreaDefinition.hh"
+#include "fastjet/JetDefinition.hh"
+#include "fastjet/tools/JetMedianBackgroundEstimator.hh"
+#include "fastjet/tools/Subtractor.hh"
+#include "fastjet/contrib/ConstituentSubtractor.hh"
+
+#include "Framework/Logger.h"
+
+enum class BkgSubMode { none,
+                        rhoAreaSub,
+                        constSub,
+                        rhoSparseSub,
```

Comment:
we should decide if this needs to be its own enum or a setter in rhoAreaSub

Reply:
I don't fully understand, you mean BkgSubMode enum should be a member in the BkgSubUtils class ?
In principle it's possible, but the problem referring to it will be a bit long `BkgSubUtils::BkgSubMode::rhoAreaSub`  instead of just using `BkgSubMode::rhoAreaSub`.